frontend/src/redux/actions/actions.tsx

```diff
@@ -147,8 +167,29 @@ export function setRouteFilter(filterFunc: (entry: Route) => Route | null) {
   return { type: UI_ROUTER_SET_ROUTE_FILTER, filterFunc };
 }
 
-export function setDetailsViewHeaderAction(actionFunc: HeaderActionType) {
-  return { type: UI_DETAILS_VIEW_SET_HEADER_ACTION, action: actionFunc };
+export function setDetailsViewHeaderAction(action: HeaderActionType | HeaderAction) {
+  let headerAction = action as HeaderAction;
+  if (headerAction.id === undefined) {
+    if (headerAction.action === undefined) {
+      headerAction = { id: '', action: action as HeaderActionType };
+    } else {
+      headerAction = { id: '', action: headerAction.action };
+    }
+  }
+  return { type: UI_DETAILS_VIEW_SET_HEADER_ACTION, action: headerAction };
+}
+
+export function addDetailsViewHeaderActionsProcessor(
+  actionProcessor: HeaderActionsProcessor | HeaderActionsProcessor['processor']
+) {
+  let headerActionsProcessor = actionProcessor as HeaderActionsProcessor;
+  if (headerActionsProcessor.id === undefined) {
+    headerActionsProcessor = {
+      id: '',
+      processor: actionProcessor as HeaderActionsProcessor['processor'],
```

Comment:
I wonder if `as` is hiding a bug.

Reply:
I'll do a type check for function so this is not needed. Thanks.